docs/use_cases/recSys-basic.ipynb

```diff
@@ -0,0 +1,3606 @@
+    "    return ids_scores[:N]\n",
+    "\n",
+    "\n",
+    "def get_similar_articles(count_id, vectors, N, count_ids_map):\n",
+    "\n",
+    "    ids_scores = get_similar_vectors(count_id, vectors, N)\n",
+    "    original_ids = [count_ids_map[a[0]] for a in ids_scores]\n",
+    "\n",
+    "    return original_ids"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "id": "a4e50d2d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def print_article_text(corpus, ids_count_map, similar_article_ids):\n",
+    "    \n",
+    "    #print(f'{N} movies/shows similar to\\n {corpus[count_id]}:\\n')\n",
+    "    for article_id in similar_article_ids:\n",
+    "        print(corpus[article_id])\n",
+    "        print('-' * 30)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71d2ffc6",
+   "metadata": {},
+   "source": [
+    "Below we will chose a context article, for which we will then search similar articles. We hope that the short artcile description would be enough to evaluate if the recommended similar articles make sense."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "id": "a0054bf8",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Context Article:  3 ‘The Opposition’ Gives Trump A Hot Lawyer Of His Own He's here to make a \"strong case\" for the president.\n"
+     ]
+    }
+   ],
+   "source": [
+    "original_id = 744\n",
+    "count_id = ids_count_map[original_id]\n",
+    "print(\"Context Article: \", count_id, corpus[count_id])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "505c633c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "k = 3 # how many similar articles to retrieve"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "id": "8b163666",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[3705, 8208, 7878]"
+      ]
+     },
+     "execution_count": 29,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "similar_articles = get_similar_articles(count_id, vectors, k, count_ids_map)\n",
+    "similar_articles"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "891e5f74",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[277, 276, 367]"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "similar_article_ids = [ids_count_map[id_] for id_ in similar_articles]\n",
+    "similar_article_ids"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "id": "ffdd8e1c",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'‘The Opposition’ Gives Trump A Hot Lawyer Of His Own He\\'s here to make a \"strong case\" for the president.'"
```

Comment:
I think it would be worth adding an example where you mix 2 articles (via element-wise mean) and search for similar articles using the aggregated vector as context. Ideally the results would relate to both initial articles that were aggregated to form the context vector.

Reply:
This can serve as a somewhat altered usecase, when the content based recommendations are also personalized